examples/example_utils.cpp

```diff
@@ -28,10 +40,17 @@ std::unique_ptr<ExampleUtils> openFromArgs(const std::string& port_or_hostname,
         if( port < 1024 || port > 65535 )
             throw std::runtime_error("Invalid TCP port (must be between 1024 and 65535.");
 
-        example_utils->connection = std::unique_ptr<mip::platform::TcpConnection>(new mip::platform::TcpConnection(port_or_hostname, port));
+#ifdef MIP_USE_EXTRAS
+        using RecordingTcpConnection = mip::extras::RecordingConnectionWrapper<mip::platform::TcpConnection>;
+        example_utils->connection = std::unique_ptr<RecordingTcpConnection>(new RecordingTcpConnection(example_utils->recordedFile.get(), example_utils->recordedFile.get(), port_or_hostname, port));
+#else  // MIP_USE_EXTRAS
+        using TcpConnection = mip::platform::TcpConnection;
+        example_utils->connection = std::unique_ptr<TcpConnection>(new TcpConnection(port_or_hostname, port));
+#endif  // MIP_USE_EXTRAS
+
         example_utils->device = std::unique_ptr<mip::DeviceInterface>(new mip::DeviceInterface(example_utils->connection.get(), example_utils->buffer, sizeof(example_utils->buffer), 1000, 2000));
 #else  // MIP_USE_TCP
```

Comment:
`-DMIP_USE_TCP` is correct when compiling the file (per the `#ifdef`s), but that define is controlled by `WITH_TCP` in cmake. So either one of these is correct, depending on whether the user is building with CMake or manually. I'd prefer leaving this (and the serial and extras equivalents) with the `MIP_USE_TCP` setting. Probably the best option would be to rename one of them to be consistent.

Reply:
They should definitely be standard. I changed the CMake flags to be `MIP_USE_*` to be consistent as otherwise that log line could be very confusing